libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -313,7 +313,7 @@ InternalDispatchMap IRGenerator::generateInternalDispatchFunctions(ContractDefin
 				solAssert(m_context.functionCollector().contains(IRNames::function(*function)), "");
 
 				cases.emplace_back(map<string, string>{
-					{"funID", to_string(*function->annotation().internalFunctionID)},
+					{"funID", to_string(*m_context.mostDerivedContract().annotation().internalFunctionIDs[function])},
```

Comment:
When getting rid of the ``SetOnce`` we can change this to
```suggestion
					{"funID", to_string(m_context.mostDerivedContract().annotation().internalFunctionIDs.at(function))},
```
which will also verify that we have an internal ID for the function and throw otherwise (https://en.cppreference.com/w/cpp/container/map/at)